### ISSUE TITLE
Modal component screwing with html injection location of Grand Child Component

### ISSUE THREAD

**Issue by jaurakunal** [dougludlow/ng2-bs3-modal#83]
Hi,

I'm facing a weird problem with the modal component. It seems to work just fine when its added to & invoked from a child component. However, when I add the modal component to a grandchild component  it seems to screw up the injection location of the grandchild html template. Instead of putting it under the router outlet of the grandchild component it puts the entire grand child html template as the last child of the Body tag.

This is causing the whole component to show up 'on top' of the rest of the page. Any help with getting this to work at a grandchild component would be much appreciated.

Cheers
Kunal

**Comment by dougludlow** [comment]
@jaurakunal interesting, can you provide a code sample/plunkr of how you're adding it to a grandchild component?

**Comment by dougludlow** [comment]
@jaurakunal are you still experiencing this problem?

**Comment by jaurakunal** [comment]
@dougludlow I'm so sorry for slacking off on updating my findings on this. I'm not seeing this error anymore.

What I realized was that the *ngIf directive was interacting in a weird way with the modal component. This was at RC4. I worked around it by using ngClass to assign a display:none; instead of using *ngIf for the time being.

I'm in the process of migrating to the GA version now. Hopefully this would get fixed with  that. I'll update this issue tracker once I finish with that.

Apologize again for the late late response on this.
